lib/CsvImportValidator.class.php

```diff
@@ -0,0 +1,466 @@
+        'utf16BigEndianBom' => self::UTF16_BIG_ENDIAN_BOM,
+        'utf32LittleEndianBom' => self::UTF32_LITTLE_ENDIAN_BOM,
+        'utf32BigEndianBom' => self::UTF32_BIG_ENDIAN_BOM,
+    ];
+    protected $context;
+    protected $dbcon;
+    protected $filenames = [];
+    protected $csvTests;
+    protected $header;
+    protected $rows = [];
+    protected $showDisplayProgress = false;
+    protected $resultCollection;
+    protected $ormClasses = [];
+
+    // Default options:
+    // Assumes QubitInformationObject if className option not set.
+    protected $validatorOptions = [
+        'className' => 'QubitInformationObject',
+        'verbose' => false,
+        'source' => '',
+        'separator' => ',',
+        'enclosure' => '"',
+        'specificTests' => '',
+        'pathToDigitalObjects' => '',
+    ];
+
+    // List of valid classNames
+    protected $defaultCsvClassNameList = [
+        'QubitInformationObject',
+        'QubitActor',
+        'QubitAccession',
+        'QubitRepository',
+        'QubitEvent',
+    ];
+
+    // General tests which can apply to any CSV.
+    protected $generalTestList = [
+        'CsvSampleValuesValidator' => CsvSampleValuesValidator::class,
+        'CsvFileEncodingValidator' => CsvFileEncodingValidator::class,
+        'CsvColumnNameValidator' => CsvColumnNameValidator::class,
+        'CsvColumnCountValidator' => CsvColumnCountValidator::class,
+        'CsvDuplicateColumnNameValidator' => CsvDuplicateColumnNameValidator::class,
+        'CsvEmptyRowValidator' => CsvEmptyRowValidator::class,
+        'CsvCultureValidator' => CsvCultureValidator::class,
+        'CsvLanguageValidator' => CsvLanguageValidator::class,
+        'CsvFieldLengthValidator' => CsvFieldLengthValidator::class,
+    ];
+
+    // Tests which pertain only to this class type.
+    protected $qubitInformationObjectTestList = [
+        'CsvParentValidator' => CsvParentValidator::class,
+        'CsvLegacyIdValidator' => CsvLegacyIdValidator::class,
+        'CsvEventValuesValidator' => CsvEventValuesValidator::class,
+        'CsvDigitalObjectPathValidator' => CsvDigitalObjectPathValidator::class,
+        'CsvDigitalObjectUriValidator' => CsvDigitalObjectUriValidator::class,
+    ];
+
+    public function __construct(sfContext $context = null,
+    $dbcon = null,
+    $options = [])
+    {
+        if (null === $context) {
+            $context = new sfContext(ProjectConfiguration::getActive());
+        }
+
+        $this->setContext($context);
+        $this->dbcon = $dbcon;
+        $this->setOptions($options);
+
+        $this->setOrmClasses(
+            [
+                'QubitFlatfileImport' => QubitFlatfileImport::class,
+                'QubitObject' => QubitObject::class,
+            ]
+        );
+    }
+
+    public function getTestsByClassType()
+    {
+        $selectedTests = $this->generalTestList;
+
+        switch ($this->validatorOptions['className']) {
```

Comment:
I have added a commit containing a bit of a refactor - I didn't really like the way the logic for building and maintaining the list of validators was spread all over inside of CsvImportValidator class. I have collected this logic into a new CsvValidatorCollection that populates itself with the correct tests given the CSV entity type specified.

The Switch statement you commented on here is gone now, since the validators themselves are responsible for declaring which entities they are limited to, if any.